listenbrainz/listenstore/influx_listenstore.py

```diff
@@ -347,6 +347,37 @@ def fetch_listens_from_storage(self, user_name, from_ts, to_ts, limit, order):
 
         return listens
 
+
+    def fetch_recent_listens_for_users(self, user_list, limit = 2):
+        """ Fetch recent listens for a list of users, given a limit which applies per user. If you 
+            have a limit of 3 and 3 users you should get 9 listens if they are available.
+
+            user_list: A list containing the users for which you'd like to retrieve recent listens.
+            limit: the maximum number of listens for each user to fetch.
+        """
+
+        escaped_user_list = []
+        for user_name in user_list:
+           escaped_user_list.append(get_escaped_measurement_name(user_name))
+
+        query = 'SELECT username, * FROM ' + ",".join(escaped_user_list) + " ORDER BY time DESC LIMIT " + str(limit)
+        try:
+            results = self.influx.query(query)
+        except Exception as err:
+            self.log.error("Cannot query influx while getting listens for users: %s: %s", user_list, str(err), exc_info=True)
+            return []
+
+        listens = []
+        for user in user_list:
+            for result in results.get_points(measurement=get_measurement_name(user)):
+                l = Listen.from_influx(result)
+                # Set the user name where the UI stuff expects it to be
+                l.data['user_name'] = user
```

Comment:
This isn't the correct place for the `user_name`.
As is, it ends up in `listen.track_metadata` instead of at the root of the `listen` object